### PR TITLE
Fixes silly siding sprites

### LINKED ISSUE #75059: Several Siding Turf Decals no longer have the correct iconstate
Caused by #74171. They're using the basic siding icons for `/end` and `/corner` rather than the specific ones they prior had.
On the plus side, I do kinda vibe with these colors being available for basic sidings too. Sometimes you just want it schmoove, yanno?

image

### LINKED ISSUE #76195: Wooden siding uses the incorrect corner and end sprites
I thought they looked a little funny, but didn't pay any mind to it.
Paid a little more mind, went to investigate it, and they're using the wrong sprites. Probably has to do with the helper automatically setting them?

This MAY apply to anything else with unique corner/end sprites, I just was looking at wood for the time being.

Wooden and regular siding decals for comparison. 
image

image

### PR DESCRIPTION
**Pull request by norsvenska**
## About The Pull Request

Detaches siding code from the macro in #74171 because it wasn't doing its job properly. I tried to fix it, but I'm not a very good coder and opted to manually define each of the variations of siding. So this is more of a bandaid fix than anything.

## Why It's Good For The Game

Fixes #76195, fixes #75059
Correct turf decals look better than wrong ones

look at how much nicer this corner looks now
image

## Changelog

:cl:
image: Siding now uses the correct sprite on corners and endpieces.
/:cl:

### CODE REVIEW

**Comment by Jolly-66** [comment]
Honestly, I was thinking of doing this but it doesn't come off to me as a proper fix, more of a temp fix to a bigger issue with the macro. I tinkered with trying to fix it but I kept running into dead ends.

Thats just my two cents.

**Comment by norsvenska** [comment]
> Honestly, I was thinking of doing this but it doesn't come off to me as a proper fix, more of a temp fix to a bigger issue with the macro. I tinkered with trying to fix it but I kept running into dead ends.

Yeah, that's how I felt. This is by no means a permanent fix, it's a bandaid one, like I said. I did try working with the macro but kept getting errors and decided to do it the long way.

**Comment by Jolly-66** [comment]
Regardless,
Bless you for fixing this my inner mapper says thank you.

**Review by Wallemations — APPROVED** [review]

File: code/game/objects/effects/decals/turfdecal/markings.dm [hunk 1]
```diff
@@ -225,170 +225,288 @@
...
-	icon_state = "siding_end";\
-}
+// /// Automatically generates all subtypes for a siding with the given path.
+// #define SIDING_SUBTYPE_HELPER(path)
+// ##path/corner{
+// 	icon_state = "siding_corner";
+// }
+// ##path/end{
+// 	icon_state = "siding_end";
+// } this is broken. also there was a backslash at the end of each line, was throwing errors when commented out. just fyi, future coder
```
Comment (san7890) [inline code/game/objects/effects/decals/turfdecal/markings.dm:235]: if it's broken, delete it. i thought it worked fine when i made it but just delete this if it's going to be completely unused

File: code/game/objects/effects/decals/turfdecal/markings.dm [hunk 2]
```diff
@@ -440,4 +558,4 @@ SIDING_SUBTYPE_HELPER(/obj/effect/turf_decal/siding/wideplating_new/terracotta)
 /obj/effect/turf_decal/board_letter/h
 	icon_state = "board_h"
 
-#undef SIDING_SUBTYPE_HELPER
+//#undef SIDING_SUBTYPE_HELPER
```
Comment (san7890) [inline code/game/objects/effects/decals/turfdecal/markings.dm:561]: delete this too

**Review by san7890 — CHANGES_REQUESTED** [review]

**Pull request merged**